cloudevents/event_v01.go

```diff
@@ -90,15 +112,16 @@ func (ec V01EventContext) AsHeaders() http.Header {
 	for k, v := range ec.Extensions {
 		data := fmt.Sprint(v)
 		encoded, err := json.Marshal(v)
-		if err != nil {
+		if err == nil {
```

Comment:
We seem to swallow the error here.
In all honesty, I have yet manage to make json.Marshal() fail, but nonetheless, at least logging the error could be quite useful.

Reply:
json.Marshal will fail if you put something non-convertible into the map[string]interface{}, like a channel.

I've changed `AsHeaders()` to return `(http.Header, error)`.